addon/mixins/in-viewport.js

```diff
@@ -91,22 +103,51 @@ export default Mixin.create({
       return;
     }
 
-    const $contextEl = $(context);
-    const boundingClientRect = element.getBoundingClientRect();
-
-    this._triggerDidAccessViewport(
-      isInViewport(
-        boundingClientRect,
-        $contextEl.innerHeight(),
-        $contextEl.innerWidth(),
-        get(this, 'viewportTolerance')
-      )
-    );
-
-    if (boundingClientRect && get(this, 'viewportUseRAF')) {
-      rAFIDS[get(this, 'elementId')] = window.requestAnimationFrame(
-        bind(this, this._setViewportEntered, context)
-      );
+    if (get(this, 'viewportUseIntersectionObserver')) {
+      const { top, left, bottom, right } = this.viewportTolerance;
+      const options = {
+        root: null, 
```

Comment:
Another question - if null, the root is the layout view (document.documentElement).  One case I have run up against before is the scrollable area is not the window but rather a confined area in the viewport.

Would you be in favor of a `scrollableArea` config variable?